Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -77,66 +82,81 @@ static bool isAllowedByFeaturePolicy(const Document& document, PermissionName na
 
 void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DOMPromiseDeferred<IDLInterface<PermissionStatus>>&& promise)
 {
-    // FIXME: support permissions in WorkerNavigator.
-
-    auto context = m_navigator ? m_navigator->scriptExecutionContext() : nullptr;
-
+    auto* context = m_navigator ? m_navigator->scriptExecutionContext() : nullptr;
     if (!context || !context->globalObject()) {
         promise.reject(Exception { InvalidStateError, "The context is invalid"_s });
         return;
     }
 
     auto* document = dynamicDowncast<Document>(context);
```

Comment:
`dynamicDowncast<Document>(*context);` would avoid an unnecessary null check

Reply:
Fixed!

Reply:
Is `if (!context || ... )` on line 86 the unnecessary null check that we are trying to avoid?

Reply:
No, otherwise I would have commented on that line. I was trying to avoid the null check done internally by `dynamicDowncast<Document>(context)` which is why I asked to change to `dynamicDowncast<Document>(*context)`. But then you made the change I suggested + dropped the earlier null checking code (which I didn't ask to do)

Reply:
Ah I see. Sorry about that--I misunderstood the issue. I'll make the fix and push again when the changes related to the observer are done as well.

Reply:
Fixed!